examples/fem_system/fem_system_ex1/naviersystem.C

```diff
@@ -267,13 +267,16 @@ bool NavierSystem::element_time_derivative (bool request_jacobian,
   for (unsigned int qp=0; qp != n_qpoints; qp++)
     {
       // Compute the solution & its gradient at the old Newton iterate
-      Number p = c.interior_value(p_var, qp),
-        u = c.interior_value(u_var, qp),
-        v = c.interior_value(v_var, qp),
-        w = c.interior_value(w_var, qp);
-      Gradient grad_u = c.interior_gradient(u_var, qp),
-        grad_v = c.interior_gradient(v_var, qp),
-        grad_w = c.interior_gradient(w_var, qp);
+      Number p, u, v, w;
+      c.interior_value(p_var, qp, p),
+      c.interior_value(u_var, qp, u),
+      c.interior_value(v_var, qp, v),
+      c.interior_value(w_var, qp, w);
+
+      Gradient grad_u, grad_v, grad_w;
```

Comment:
> Out of curiosity, did you guys have profiled it ? 

Definitely didn't profile it, and doubt that it makes any performance difference in practice, but IMO the old API should be preferred if you want to declare the variables inside the loop since it combines creation and initialization into a single statement. The current version is the worst of both worlds since it both opens up the possibility to forget to set the values *and* does separate creation and initialization steps.